tests/epyccel/test_epyccel_classes.py

```diff
@@ -501,3 +501,25 @@ def test_class_globals_visitation_order(language):
     a_l = modnew.A()
 
     assert a_py.x == a_l.x
+
+def test_class_variables(language):
+    import classes.class_variables as mod
+    modnew = epyccel(mod, language = language)
+
+    a_py = mod.A(True)
+    a_l = modnew.A(True)
+
+    assert a_py.x == a_l.x
+    assert a_py.z == a_l.z
+    assert a_py.get_4() == a_l.get_4()
+    assert np.array_equal(a_py.y, a_l.y)
+    assert a_py.get_y_len() == a_l.get_y_len()
+
+    a_py = mod.A(False)
+    a_l = modnew.A(False)
+
+    assert a_py.x == a_l.x
+    assert a_py.z == a_l.z
+    assert a_py.get_4() == a_l.get_4()
+    assert np.array_equal(a_py.y, a_l.y)
+    assert a_py.get_y_len() == a_l.get_y_len()
```

Comment:
How about a parametric unit test?

```suggestion
@pytest.mark.parametrize('bool_param', [True, False])
def test_class_variables(language, bool_param):
    import classes.class_variables as mod
    modnew = epyccel(mod, language = language)

    a_py = mod.A(bool_param)
    a_l = modnew.A(bool_param)

    assert a_py.x == a_l.x
    assert a_py.z == a_l.z
    assert a_py.get_4() == a_l.get_4()
    assert np.array_equal(a_py.y, a_l.y)
    assert a_py.get_y_len() == a_l.get_y_len()
```

Reply:
Code duplication reduced: [175e225](https://github.com/pyccel/pyccel/pull/2434/commits/175e225d461c6ed3bd5b81d642e7abb7ce4fce58)